packages/bot/src/scratch/hooks/toolbox.js

```diff
@@ -80,122 +80,135 @@ Blockly.Toolbox.prototype.populate_ = function (newTree) {
 
+        is_search = true;
+
         if (search_term.length <= 1) {
             flyout.setVisibility(false);
             return;
         }
 
-        flyout_content = {
-            type  : 'search',
-            blocks: [],
-        };
+        const { derivWorkspace: { toolbox_: { categoryMenu_: { categories_ } } } } = Blockly;
+        // eslint-disable-next-line consistent-return
+        const block_contents = categories_
+            .filter(category => !category.has_child_category_)
+            .map(category => {
+                let contents = category.getContents();
 
-        if (typeof search_term === 'string') {
-            search_term = search_term.trim().toLowerCase();
-            search_term = search_term.split(' ');
-        }
+                if (typeof contents === 'string') {
+                    const fnToApply = Blockly.derivWorkspace.getToolboxCategoryCallback(contents);
+                    contents = fnToApply(Blockly.derivWorkspace);
+                }
 
-        const blocks = Blockly.Blocks;
-        Object.keys(blocks).forEach(blockKey => {
-            let keywords = ` ${blockKey}`;
-            const block = blocks[blockKey];
+                const only_block_contents = contents.filter(content => content.tagName.toUpperCase() === 'BLOCK');
+                return only_block_contents;
+            })
+            .flat();
+
+        block_contents.forEach(block_content => {
+            const block_type = block_content.getAttribute('type');
+            const block = Blockly.Blocks[block_type];
             const block_meta = block.meta instanceof Function && block.meta();
-            const block_definition = block.definition instanceof Function && block.definition();
+            const block_definitions = block.definition instanceof Function && block.definition();
 
             if (!block_meta) {
                 return;
             }
+            
+            // block_name matched
+            if (block_type.toUpperCase().includes(search_term)) {
+                flyout_content.unshift(block_content);
+                return;
+            }
 
-            Object.keys(block_meta).forEach(key => {
-                const meta = block_meta[key];
-                keywords += ` ${meta}`;
-            });
+            // block_meta matched
+            const matched_meta = Object.keys(block_meta)
+                .find(key => block_meta[key].toUpperCase().includes(search_term));
+            if (matched_meta && matched_meta.length) {
+                flyout_content.push(block_content);
+                return;
+            }
 
-            Object.keys(block_definition).forEach(key => {
-                const definition = block_definition[key];
+            // block_definition matched
+            const definition_key_to_search = /^((message)|(tooltip)|(field_dropdown$))/;
+            // eslint-disable-next-line consistent-return
```

Comment:
We don't need this eslint exception here as you're `return`ing `false` by default.